internal/pagination/pagination.go

```diff
@@ -0,0 +1,53 @@
+package pagination
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/rs/zerolog"
+)
+
+type ctxKeyType int
+
+const (
+	defaultValueLimit             = 100
+	defaultValueOffset            = 0
+	offsetCtxKey       ctxKeyType = iota
+	limitCtxKey
+)
+
+// WithOffset returns context copy with offset value.
+func WithOffset(ctx context.Context, offsetStr string) context.Context {
+	logger := zerolog.Ctx(ctx)
+	offset, err := strconv.Atoi(offsetStr)
+	if err != nil || offset < 0 {
+		logger.Trace().Msg("Offset is missing, setting offset value to 0")
+		offset = defaultValueOffset
+	}
+	return context.WithValue(ctx, offsetCtxKey, offset)
+}
+
+// WithLimit returns context copy with limit value.
+func WithLimit(ctx context.Context, limitStr string) context.Context {
+	logger := zerolog.Ctx(ctx)
+	limit, err := strconv.Atoi(limitStr)
+	if err != nil || limit < 0 {
+		logger.Trace().Msg("Limit is missing, setting offset value to 0")
+		limit = defaultValueLimit
+	}
+	return context.WithValue(ctx, limitCtxKey, limit)
+}
+
+func GetOffset(ctx context.Context) int {
+	if offset, ok := ctx.Value(offsetCtxKey).(int); ok {
+		return offset
+	}
+	return defaultValueOffset
+}
+
+func GetLimit(ctx context.Context) int {
```

Comment:
```suggestion
func Limit(ctx context.Context) int {
```

Get prefix for getters is considered an anti-pattern in Go.